src/wallet/wallet.cpp

```diff
@@ -3950,8 +3952,13 @@ ScriptPubKeyMan* CWallet::AddWalletDescriptor(WalletDescriptor& desc, const Flat
         return nullptr;
     }
 
+    std::string error;
     auto spk_man = GetDescriptorScriptPubKeyMan(desc);
     if (spk_man) {
+        if (!spk_man->CanUpdateToWalletDescriptor(desc, error)) {
+            WalletLogPrintf("Invalid param: %s\n", error);
+            throw JSONRPCError(RPC_INVALID_PARAMETER, error);
```

Comment:
Shouldn't return null as other validations like the previous one (`IsWalletFlagSet`)?
```suggestion
            return nullptr;
```
Moreover, `DescriptorScriptPubKeyMan::UpdateWalletDescriptor` also calls `DescriptorScriptPubKeyMan::CanUpdateToWalletDescriptor`... could this call be removed?

Reply:
your right CanUpdateToWalletDescriptor removed completely no need now. I've used UpdateWalletDescriptor only.